cmd/cloud-controller-manager/.import-restrictions

```diff
@@ -1,4 +1,37 @@
 rules:
   - selectorRegexp: k8s[.]io/kubernetes
     allowedPrefixes:
-      - k8s.io/kubernetes/cmd/cloud-controller-manager
\ No newline at end of file
+      - k8s.io/kubernetes/pkg/api/legacyscheme
```

Comment:
was adding these back intentional or a bad rebase?

Reply:
intentional. This is indirectly imported by `k8s.io/kubernetes/pkg/controller/testutil` -> `node_ipam_controller_test` in `k8s.io/kubernetes/pkg/controller/nodeipam'.

Reply:
I could have a follow up PR to put the samples(like nodeIpamController) in its own directory and also split out the base/vanilla version from the sample showing usage of the extensions as mentioned above. In this way the .import-restrictions file might look better and make less confusion for cloud providers to follow.

Reply:
The changes under staging look fine to me.

I'm having trouble understanding the current status/consumers and goals for each of the following:

1. current `cloud-controller-manager` command / binary (cmd/cloud-controller-manager)
  * is this included in current releases or image manifests built/pushed? are there consumers of this today?
2. ideal minimal example `main()`
  * I expected this to live in staging `k8s.io/cloud-provider` with no k8s.io/kubernetes dependencies
3. an example that stitches in node_ipam to the minimal example (either by moving out node_ipam or by living in k8s.io/kubernetes)

Modifying the existing `main()` to add node_ipam without first extracting the minimal k8s.io/kubernetes-free one is making the path to the minimal on harder to see. Is that still the plan?

Reply:
1.a) As a fast follow it might be nice to split the binary. Have a vanilla binary with none of the extensions in use which a cloud provider could copy and modify to get started. A sample which shows how to use each of the extension mechanisms. I think the vanilla should be here and the sample under a sample sub directory?

1.b) The CCM in K/K is not included in any releases but it is built to ensure it remains buildable. It is forked and modified into various other repos where the modified version is released. These changes should allow the cloud providers to just depend on libraries, no more forking needed. 

2.a) +1, see the fast follow suggestion above. 

2.b) +Alot, if we break the sample into its own directory, that should allow the base version to have no K8s.io/kubernetes dependencies. More importantly we can then enforce that the base version have no K8s.io/kubernetes dependencies.

3) +1